### ISSUE TITLE
Make Project ID optional when using machine ID with `.infisical.json`

### ISSUE THREAD

**Issue by edishu** [Infisical/infisical#1850]
### Feature description
File `.infisical.json` contains project ID. However, when starting application using machine ID project ID is required to be added to startup command.

It is redundant to provide project ID twice and can result in drift between local and production development environments.

### Why would it be useful?
If machine ID logged in user can use `.infisical.json` just like a person user it will make development easier and eliminate possibility of drift between local and production builds.

### Additional context
Application startup command throws error when no project ID is provided:

```
Project ID is required when using machine identity
```

**Comment by ohmydevops** [comment]
Same

**Comment by en3sis** [comment]
Yes please!

**Comment by Arnaud-IFEA** [comment]
Hello,
I had a hard time understanding what i was doing wrong... until i realize that machine identity does not take .infisical.json into account.
Why is that, i really don't understand ?

**Comment by sjacobflaherty** [comment]
Same

**Comment by adrianipopescu** [comment]
yeah the docs don't help with this, https://infisical.com/docs/cli/commands/export#infisical-export:projectid

**Comment by MrChrisRodriguez** [comment]
Furthmore, `INFISICAL_TOKEN` is read from the envvars. Why isn't `INFISICAL_PROJECT_ID` as well?

IMO, logical coalescing would check, in order:
1. --projectId
2. shell env
3. infisical.json

**Comment by drtobbyas** [comment]
Still suprised this is still an open issue since 2024

**Comment by adrianipopescu** [comment]
I'd add folder and environment (think dev, stage, prod, prod-pr124) as well, my goal is to be able to have that file in the ci, pass secrets for authentication to infisical server, everything else can be the json

especially with the pattern of having multiple dev projects in the secrets project to allow secret reuse/referencing

bonus points for infisical server url there as well

bonus bonus points for supporting interpolation with a mustache type schema (or whatever kids use nowadays) with env for versatile / override scenarios, ex: have a main file, create new feature branch, pass the override as an env to the pipeline, pretty much allowing for all of the above to be a variable

**Comment by HerrNiklasRaab** [comment]
Must be really hard to implement this feature

**Comment by adrianipopescu** [comment]
no need to drop these sort of lines

if you want you can open a pr

**Comment by pablodecm** [comment]
Using INFISICAL_PROJECT_ID works now.
